packages/components/Avatar/src/useAvatar.ts

```diff
@@ -43,6 +46,19 @@ export const useAvatar = (props: AvatarProps): AvatarInfo => {
     showBadge,
   };
 
+  let imageProps: ImageProps = {
+    accessibilityLabel,
+    source: imageUrl ? ({ uri: imageUrl } as ImageSourcePropType) : undefined,
+  };
+
+  if (image && image.source) {
+    const { source, ...restImageProps } = image;
```

Comment:
What's the reason for splitting out source here? The code doesn't seem to do anything additional with source.

Reply:
yeah, I'll just reassign it. I added a check for the source later. In this case it's not needed to add it separately